front/src/modules/ui/table/components/TableColumnDropdownMenu.tsx

```diff
@@ -52,6 +53,19 @@ export const TableColumnDropdownMenu = ({
     handleColumnVisibilityChange(column);
   };
 
+  const handleEscapeKey = (event) => {
+    if (event.key === 'Escape') {
+      closeDropdownButton();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleEscapeKey);
+    return () => {
+      document.removeEventListener('keydown', handleEscapeKey);
+    };
+  }, []);
```

Comment:
Hi @Parvezkhan0, thank you for contributing!

We have a hook for listening for keys: `useScopedHotkeys`. For reference, you can see how we use this hook to listen to the "Escape" key in `TableOptionsDropdownContent`.

Please note that the third argument for `useScopedHotkeys` is what we call a "HotkeyScope": it prevents conflicts between key listeners from different components.

For this third argument, you can just create a new enum entry in `front/src/modules/ui/table/types/TableHotkeyScope.ts`. For instance, you could name it `ColumnDropdown`. Then you can pass `TableHotkeyScope.ColumnDropdown` to the `useScopedHotkeys` hook!